### PR TITLE
Update setup

### PR DESCRIPTION
**Pull request by gcperk**
Hey @ateucher , 
I updated PEMprepr and PEMr to rework the format and continued to add more functions. I tracked my key notes/questions and things to consider [here](https://github.com/ninoxconsulting/PEMprepr/issues/6). I checked with @ccarmour's workflow script to incorporate any sticky pts, although these seem mainly to be with checking input/output and file structure, so i think we have most of these covered off now. 
Key updates:
- converted vignette back to .RMD and move to PEMr package with a skeleton pipeline/workflow (still to be filled out in detail). 
- convert core templates back to .R and added scripts to show functions as placeholders in order
- added functions: create_vectors, create_raster_template, create_covariates and get_cded_dem. 
- added placeholder tests for the above. 
- I left a placeholder for @ccarmour's lidar processing script updates to be added when complete.

It would be great to get your input on these two pull requests before i get too far down the rabbit hole.
Thanks! 

@ccarmour and @whmacken - tagging you both to keep you in the loop, no need for input at this time.

### CODE REVIEW

File: R/create_base_vectors.R [hunk 1]
```diff
@@ -6,294 +6,416 @@
...
+
+  # # 1) BEC Biogeographical linework
+  bec <- bcdata::bcdc_query_geodata("f358a53b-ffde-4830-a325-a5a03ff672c3") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select("MAP_LABEL") |>
+    bcdata::collect() |>
+    dplyr::select("MAP_LABEL") %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
```
Comment (ateucher) [inline R/create_base_vectors.R:95]:
```suggestion
    dplyr::select("MAP_LABEL")
  
  if (nrow(bec) > 0) {
    bec <- sf::st_intersection(bec, aoi)
  }
```

Simplify: take it out of pipeline so avoid confusing `.` usage. What should happen when nrow == 0? The user should be alerted?

File: R/create_base_vectors.R [hunk 2]
```diff
@@ -6,294 +6,416 @@
...
+  tem <- bcdata::bcdc_query_geodata("0a83163b-a62f-4ce6-a9a1-21c228b0c0a3") %>%
+    bcdata::filter(bcdata::INTERSECTS(aoi)) %>%
+    bcdata::collect() %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
+
+  if (nrow(tem) > 0) {
+    sf::st_write(tem, fs::path(out_dir, "tem.gpkg"), append = FALSE)
+  }
```
Comment (ateucher) [inline R/create_base_vectors.R:219]:
```suggestion
    bcdata::collect()

  if (nrow(tem) > 0) {
    tem <- sf::st_intersection(tem, aoi)
    sf::st_write(tem, fs::path(out_dir, "tem.gpkg"), append = FALSE)
  }
```

Same suggestion as in `get_BEC()`, however one question - what happens when `nrow(tem) == 0`? Right now it returns nothing in that case, with no message or warning...

File: R/create_base_vectors.R (hunk 1 above)
Comment (ateucher) [reply R/create_base_vectors.R:95]: This pattern applies to `get_TEM()`, `get_VRI()`, `get_fires()`, `get_harvest()`, `get_parks()`, `get_roads()`, `get_transmission_lines()`

File: R/create_base_vectors.R [hunk 3]
```diff
@@ -6,294 +6,416 @@
...
+  )
+  # post process VRI data into classes:
+
+  # Depending on the study area we want to focus on sampling in older areas - class 4 (60-80) or 5 (80 + )
+  # To do this we will define 2 vri age classess to exclude from the sampling area
+  # class 1 and 2 (0 - 40 yrs)
+  # class 1-3 (0 - 60 years)
+
+  vri_class2 <- vri |>
+    dplyr::mutate(age_class = as.numeric(PROJ_AGE_CLASS_CD_1)) |>
```
Comment (ateucher) [inline R/create_base_vectors.R:136]: When you run `check()` you get lots of "no visiible binding for global variable". Unquoted variable names like these should be used with the `.data$` pronoun:

```suggestion
    dplyr::mutate(age_class = as.numeric(.data$PROJ_AGE_CLASS_CD_1)) |>
```

And in some package-level documentation (e.g., if you run `usethis::use_package_doc()`, you need to add:

```r
#' @importFrom rlang .data
```

File: R/create_covariates.R [hunk 4]
```diff
@@ -0,0 +1,1264 @@
...
+  #  saga_path = "C:\\Programs\\saga-9.2.0_x64\\saga-9.2.0_x64"
+  #  out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel
+  #  layers = "all"
+  #  # end testing
+
+
+  # internal data for package
+    data("layer_options", envir=environment())
+    data("moddir", envir=environment())
+    data("artifacts", envir=environment())
```
Comment (ateucher) [inline R/create_covariates.R:61]: Where are these datasets coming from? I see `R/sysdata.rda`, but don't see how it's created. Can we have a `data-raw` folder containing scripts/source data to reproducibly create it?

File: R/create_base_vectors.R [hunk 5]
```diff
@@ -6,294 +6,416 @@
...
+  water_records <- c(
+    "cb1e3aba-d3fe-4de1-a2d4-b8b6650fb1f6", # lakes
+    "f7dac054-efbf-402f-ab62-6fc4b32a619e", # rivers
+    "93b413d8-1840-4770-9629-641d74bd1cc6"
+  ) # wetlands
+  for (i in 1:length(water_records)) {
+    waterbodies <- bcdata::bcdc_query_geodata(water_records[i]) %>%
+      bcdata::filter(bcdata::INTERSECTS(aoi)) %>%
+      bcdata::collect() %>%
+      dplyr::select(id, WATERBODY_TYPE, AREA_HA)
```
Comment (ateucher) [inline R/create_base_vectors.R:236]:
```suggestion
      dplyr::select("id", "WATERBODY_TYPE", "AREA_HA")
```

Variable names in `select()` should be quoted

Comment (ateucher) [reply R/create_base_vectors.R:236]: See https://dplyr.tidyverse.org/articles/in-packages.html#data-masking-and-tidy-selection-notes

File: R/create_base_vectors.R (hunk 3 above)
Comment (ateucher) [reply R/create_base_vectors.R:136]: See https://dplyr.tidyverse.org/articles/in-packages.html#data-masking-and-tidy-selection-notes

File: R/create_covariates.R [hunk 6]
```diff
@@ -0,0 +1,1264 @@
...
+#'
+#' @param dtm a character or file path location to raster with elevation data.
+#'        Outputs will be produced at the same resolution and extent as input dtm
+#' @param saga_path a character of the file to the SAGA directory on the analysts system.
+#'          On linux systems with SAGA GIS installed Use `SAGApath = ""`
+#' @param out_dir file.path. Directory where covariates will be written. If not
+#'     specified uses the default from the `fid` folder structure. If files already
+#'     exists they will NOT be overwritten.
+#' @param layers character vector. Covariates to be created. Default is \code(`all`)
+#'  - call \code(`PEMprepr::layer_options`) for full list.
```
Comment (ateucher) [inline R/create_covariates.R:16]:
```suggestion
#' @param layers character vector. Covariates to be created. Default is `"all"`
#'  - call `PEMprepr::layer_options` for full list.
```

**Review by ateucher — CHANGES_REQUESTED** [review]
Hey @gcperk since this is a pretty big PR, I suggest we do two rounds of review. This one is mostly structural, if all of these comments are addressed, it should pass R CMD check. After that I can do a second review for the content/functionality?

There is a mix of the new base pipe `|>` and the magrittr pipe `%>%`. I suggest we stick to just one. The base pipe was introduced in R 4.1, so if we want to use it we need to update the minimum R version in `DESCRIPTION`.

File: R/create_base_vectors.R [hunk 7]
```diff
@@ -6,294 +6,416 @@
...
+#' create_base_vectors(sn_aoi, out_dir = PEMprepr::read_fid()$dir_1010_vector$path_abs)
+#'}
+create_base_vectors <- function(aoi = fs::path(PEMprepr::read_fid()$dir_1010_vector$path_abs, "aoi_snapped.gpkg"),
+                                out_dir = PEMprepr::read_fid()$dir_1010_vector$path_abs) {
+  # # testing
+  # aoi <- fs::path(PEMprepr::read_fid()$dir_1010_vector$path_abs, "aoi_snapped.gpkg")
+  #  aoi <- sf::st_read(aoi, quiet = TRUE)
+  #  out_dir <- PEMprepr::read_fid()$dir_1010_vector$path_abs
+
+  if (missing(aoi)) stop("'aoi' is missing with no default")
```
[the remark this answers is not in the document]
Comment (ateucher) [reply R/create_base_vectors.R:26]: PS: Happy to defer this comment to round 2

File: DESCRIPTION [hunk 8]
```diff
@@ -21,11 +21,20 @@ BugReports: https://github.com/ninoxconsulting/PEMprepr/issues
...
     fs,
-    sf
+    sf,
+    terra,
+    bcdata,
+    dplyr,
+    PEMr,
+    magrittr
+Remotes:
+    ninoxconsulting/PEMr
```
[the remark this answers is not in the document]
Comment (ateucher) [reply DESCRIPTION:31]: Since we don't depend on it any more

File: DESCRIPTION [hunk 9]
```diff
@@ -21,11 +21,20 @@ BugReports: https://github.com/ninoxconsulting/PEMprepr/issues
 Imports: 
     cli,
     fs,
-    sf
+    sf,
+    terra,
+    bcdata,
+    dplyr,
+    PEMr,
```
[the remark this answers is not in the document]
Comment (gcperk) [reply DESCRIPTION:28]: I suspect we will use purr and stringr again (but could also be base). I like the idea do suggests, given we only make this call once.

File: R/utils-pipe.R
[the remark this answers is not in the document]
Comment (gcperk) [reply R/utils-pipe.R:1]: I used the %>% pipe as I found the new base option was not correctly working for one of the functions. Do you have a clean suggestion for this. I can go back and flag the function where it was not working.

File: R/create_covariates.R (hunk 4 above)
Comment (gcperk) [reply R/create_covariates.R:61]: This code was developed by someone else, so I'm not 100% sure I want to keep this format. They function as a filter and check for anomalies when covariates are created by tiles (ie some covariates are dependent on larger scales than a tile and artifacts are created when tiling). 

Perhaps we can discuss this one in more detail. I don't like have undocumented data creation, however they were created from a csv conversion to .R (I suspect).

File: R/create_covariates.R (hunk 6 above)
Comment (gcperk) [reply R/create_covariates.R:16]: Will keep this for now, but might depend on how we rework the create_covariates().

File: R/create_base_vectors.R (hunk 1 above)
Comment (gcperk) [reply R/create_base_vectors.R]: updates these to same format. I think adding a flag is a good next step where n ==0. I will add these in stage 2.

File: R/create_base_vectors.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (gcperk) [reply R/create_base_vectors.R:26]: Keep this for round 2

File: R/create_base_vectors.R (hunk 5 above)
Comment (gcperk) [reply R/create_base_vectors.R]: updated

**Comment by gcperk** [comment]
hi @ateucher, I like your idea to split this into two reviews. I have made the requested changes to get the rmd check to work. Although there is still an undefined variable "data" in the utils.R . I think this function would require a bit more thought as it also relates to the .rda files. Perhaps a neater way to compact this. I am passing this back to you with a few notes on outstanding tasks for me to complete. Thanks again for your input.

File: R/create_base_vectors.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (ateucher) [reply R/create_base_vectors.R]: I've made S3 generic + methods for this; happy to discuss more @gcperk

**Pull request merged**